casper/src/test/scala/coop/rchain/casper/batch1/MergeabilityRules.scala

```diff
@@ -1,484 +0,0 @@
-            )
-            mergedState <- DagMerger.merge[Task](
-                            dag,
-                            bBlock.blockHash,
-                            baseCheckpoint.root,
-                            indices(_).deployChains.pure[Task],
-                            historyRepo
-                          )
-            mergedRoot        = mergedState._1
-            rejectedDeployOpt = mergedState._2.headOption
-            referenceState = rejectedDeployOpt
-              .map { v =>
-                List((leftDeploy, b1State), (rightDeploy, b2State))
-                  .filter {
-                    case (deploy, _) => deploy.deploy.sig != v
-                  }
-                  .map { case (_, refState) => refState }
-                  .head
-              }
-              .getOrElse(b1State)
-
-            dataContinuationAtBaseState <- getDataContinuationOnChannel0(
-                                            runtime,
-                                            baseCheckpoint.root
-                                          )
-            dataContinuationAtMainState <- getDataContinuationOnChannel0(
-                                            runtime,
-                                            leftCheckpoint.root
-                                          )
-            dataContinuationAtMergingState <- getDataContinuationOnChannel0(
-                                               runtime,
-                                               rightCheckpoint.root
-                                             )
-            dataContinuationAtMergedState <- getDataContinuationOnChannel0(
-                                              runtime,
-                                              mergedRoot
-                                            )
-            _ = println(s"conflicts found: ${mergedState._2.size}")
-            _ = println(s"base state: ${dataContinuationAtBaseState}")
-            _ = println(s"main state: ${dataContinuationAtMainState}")
-            _ = println(s"merging state: ${dataContinuationAtMergingState}")
-            _ = println(s"merged state: ${dataContinuationAtMergedState}")
-            _ = println(s"reference state: ${referenceState}")
-
-            errMsg = s""" FAILED
-                        | base = ${base.value}
-                        | b1   = ${b1.value}
-                        | b2   = ${b2.value}
-                        | Conflict: ${rejectedDeployOpt.isDefined} should be ${isConflict}
-                        | Merged state
-                        | ${dataContinuationAtMergedState}
-                        | should be
-                        | ${referenceState}
-                        |
-                        | go see it at ${file.value}:${line.value}
-                        | """.stripMargin
-            _ <- Sync[Effect]
-                  .raiseError(new Exception(errMsg))
-                  .whenA(dataContinuationAtMergedState != referenceState)
```

Comment:
@nzpr Codes only check the state. Never check if they are conflict or not

Reply:
but this effectively also checks conflict? if conflict detected wrong - state will be wrong, do I miss something?

Reply:
The name `MergeableCase` and `ConflictCase` don't make any difference.The test codes only check the state without caring about if they are conflict or not.

look at this pr https://github.com/rchain/rchain/pull/3447/files and the ci tests.

Reply:
I believe they are different because `checkConflictAndMergeStatete` has `isConflict` field which is different. This is not checked as a test case, but printed when there is state mismatch, so you can see whether the problem is in wrong conflict detection.

Reply:
> This is not checked as a test case
why not check it?

Reply:
In short, is it wrong to check it in the test?

I believe the tests are more robust with this additional check, aren't they?

And the result (in https://github.com/rchain/rchain/pull/3447/files) turns out to be something is not as expectations.

Reply:
I agree that this check makes sense, but are you saying that when you added 
```
 <- Sync[Effect]
                  .raiseError(new Exception(errMsg))
                  .whenA(rejectedDeployOpt.isDefined != isConflict)
```
this were failing?

I'm asking all this because I still don't understand what was wrong with regards to <https://github.com/rchain/rchain/pull/3430#discussion_r667340464>, trying to figure out why tests were passing when they should be not apparently?

Reply:
I am sorry that I mislead you somewhere else here.

The comment here is actually not related to https://github.com/rchain/rchain/pull/3430#discussion_r667340464 ( I thought they were related.  My bad.)

I put another explanation on that comment below. Does that make sense? -> https://github.com/rchain/rchain/pull/3430#discussion_r667421501